<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Tao Yang xadmyangt@gmail.com" />


<title>Evaluate reproducibility of Hi-C data with hicrep</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">Evaluate reproducibility of Hi-C data with <code>hicrep</code></h1>
<h4 class="author">Tao Yang <a href="mailto:xadmyangt@gmail.com">xadmyangt@gmail.com</a></h4>
<h4 class="date">2/24/2020</h4>



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Hi-C data analysis and interpretation are still in their early stages. In particular, there has been a lack of sound statistical metric to evaluate the quality of Hi-C data. When biological replicates are not available, investigators often rely on eithervisual inspection of Hi-C interaction heatmap or examining the ratio of long-range interaction read pairs over the total sequenced reads, neither of which are supported by robust statistics. When two or more biological replicates are available, it is a common practice to compute either Pearson or Spearman correlation coefficients between the two Hi-C data matrices and use them as a metric for quality control. However, these kind of over-simplified approaches are problematic and may lead to wrong conclusions, because they do not take into consideration of the unique characteristics of Hi-C data, such as distance-dependence and domain structures. As a result, two un-related biological samples can have a strong Pearson correlation coefficient, while two visually similar replicates can have poor Spearman correlation coefficient. It is also not uncommon to observe higher Pearson and Spearman correlations between unrelated samples than those between real biological replicates.</p>
<p>we develop a novel framework, <code>hicrep</code>, for assessing the reproducibility of Hi-C data. It first minimizes the effect of noise and biases by smoothing Hi-C matrix, and then addresses the distance-dependence effect by stratifying Hi-C data according to their genomic distance. We further adopt a stratum-adjusted correlation coefficient (SCC) as the measurement of Hi-C data reproducibility. The value of SCC ranges from -1 to 1, and it can be used to compare the degrees of differences in reproducibility. Our framework can also infer confidence intervals for SCC, and further estimate the statistical significance of the difference in reproducibility measurement for different data sets.</p>
<p>In this Vignette, we explain the method rationale, and provide guidance to use the functions of <code>hicrep</code> to assess the reproducibility for Hi-C intrachromosome replicates.</p>
</div>
<div id="citation" class="section level2">
<h2>Citation</h2>
<p>Cite our paper:</p>
<p>HiCRep: assessing the reproducibility of Hi-C data using a stratum-adjusted correlation coefficient. Tao Yang, Feipeng Zhang, Galip Gurkan Yardimci, Fan Song, Ross C Hardison, William Stafford Noble, Feng Yue, Qunhua Li. Genome Research 2017. doi: 10.1101/gr.220640.117.</p>
</div>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p>Download the source package (hicrep_xxx.tar.gz) from Github. Or install it from Bioconductor:</p>
<pre><code>## try http:// if https:// URLs are not supported
if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)

BiocManager::install(&quot;hicrep&quot;)</code></pre>
</div>
<div id="rationale-of-method" class="section level2">
<h2>Rationale of method</h2>
<p>This is a 2-step method (Figure1). In Hi-C data it is often difficult to achieve sufficient coverage. When samples are not sufficiently sequenced, the local variation introduced by under-sampling can make it difficult to capture large domain structures. To reduce local variation, we first smooth the contact map before assessing reproducibility. Although a smoothing filter will reduce the individual spatial resolution, it can improve the contiguity of the regions with elevated interaction, consequently enhancing the domain structures. We use a 2D moving window average filter to smooth the Hi-C contact map. This choice is made for the simplicity and fast computation of mean filter, and the rectangular shape of Hi-C compartments.</p>
<p>In the second step, we stratify the Hi-C reads by the distance of contacting loci, calculate the Pearson correlations within each stratum, and then summarize the stratum-specific correlation coefficients into an aggregated statistic. We name it as Stratum-adjusted Correlation Coefficient (SCC). For the methodology details, please refer to our paper on Genome Research.</p>
<p>[Figure1. <code>hicrep</code> pipeline schematic representation] (<a href="https://raw.githubusercontent.com/MonkeyLB/hicrep/master/vignettes/hicrep-pipeline.JPG" class="uri">https://raw.githubusercontent.com/MonkeyLB/hicrep/master/vignettes/hicrep-pipeline.JPG</a>)</p>
</div>
<div id="the-format-of-input-and-pre-processing" class="section level2">
<h2>The format of input and Pre-processing</h2>
<p>The input are two Hi-C matrices to be compared. The Hi-C matrices should have the dimension <span class="math inline">\(N\times N\)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(HiCR1)</code></pre></div>
<pre><code>## [1] 52 52</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">HiCR1[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</code></pre></div>
<pre><code>##    V4 V5 V6 V7 V8 V9 V10 V11 V12 V13
## 1   0  0  0  0  0  0   0   0   0   0
## 2   0  0  0  0  0  0   0   0   0   0
## 3   0  0  0  0  0  0   0   0   0   0
## 4   0  0  0  0  0  0   0   0   0   0
## 5   0  0  0  0  0  0   0   0   0   0
## 6   0  0  0  0  0  0   0   0   0   0
## 7   0  0  0  0  0  0   0   0   0   0
## 8   0  0  0  0  0  0   0   0   0   0
## 9   0  0  0  0  0  0   0   0   0   0
## 10  0  0  0  0  0  0   0   0   0   0</code></pre>
<p>The function <code>get.scc</code> will first smooth the HiC matrix, with given neighborhood size parameter <span class="math inline">\(h\)</span>, and filter the bins that have zero counts in both replicates. The arguments includes the two matrices, the resolution of matrices, smoothing parameter, and the lower bound and upper bound of interaction distance considered. The resolution is simply the bin size. Smoothing parameter decides the neighborhood size of smoothing. Below (Figure 2) is a representation of smoothing neighborhood for a point <span class="math inline">\(C_{ij}\)</span>:</p>
<p>[Figure2. A smoothing neighborhood with <span class="math inline">\(h = 2\)</span>] (<a href="https://raw.githubusercontent.com/MonkeyLB/hicrep/master/vignettes/Smoothing_neighborhood.JPG" class="uri">https://raw.githubusercontent.com/MonkeyLB/hicrep/master/vignettes/Smoothing_neighborhood.JPG</a>)</p>
</div>
<div id="calculate-stratum-adjusted-correlation-coefficient-scc" class="section level2">
<h2>Calculate Stratum-adjusted Correlation Coefficient (SCC)</h2>
<p>An example to calculate SCC for a matrix of 1Mb resolution. Smoothing parameter <span class="math inline">\(h\)</span> is set to 2. The lower bound of distance considered is 0 (diagnal), and the upper bound is 5Mb.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scc.out =<span class="st"> </span><span class="kw">get.scc</span>(HiCR1, HiCR2, <span class="dv">1000000</span>, <span class="dv">2</span>, <span class="dv">0</span>, <span class="dv">5000000</span>)

<span class="co">#SCC score</span>
scc.out<span class="op">$</span>scc</code></pre></div>
<pre><code>##           [,1]
## [1,] 0.9865779</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Standard deviation of SCC</span>
scc.out<span class="op">$</span>std</code></pre></div>
<pre><code>## [1] 0.0009243735</code></pre>
<p>The output is a list of results including stratum specific Pearson correlations, weight coefficient, SCC and the asymptotic standard deviation of SCC. The last two numbers are the ones we needed in most of the situations.</p>
</div>
<div id="smooth-the-hi-c-matrix-with-2d-mean-filter" class="section level2">
<h2>Smooth the Hi-C matrix with 2D mean filter</h2>
<p>The function <code>fast.mean.filter()</code> is a very fast algorithm that applies 2D mean filter to squred matrices such as Hi-C contact mapes. The output is a smoothed matrix that has the same size with the original matrix. Here is an example to smooth the matrix with parameter <span class="math inline">\(h = 2\)</span>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">smd_mat =<span class="st"> </span><span class="kw">fast.mean.filter</span>(HiCR1, <span class="dv">2</span>)</code></pre></div>
</div>
<div id="select-the-optimal-smoothing-parameter" class="section level2">
<h2>Select the optimal smoothing parameter</h2>
<p>To select <span class="math inline">\(h\)</span> objectively, we develop a heuristic procedure to search for the optimal smoothing parameter. Our procedure is designed based on the observation that the correlation between contact maps of replicate samples first increases with the level of smoothness and plateaus when sufficient smoothness is reached.To proceed, we use a pair of reasonably deeply sequenced interaction map as the training data. We randomly sampled 10% of the data, then compute SCC for the sampled dataeach fraction at a series of smoothing parameters in the ascending order. We choose the smallest <span class="math inline">\(h\)</span> at which the increment of the average reproducibility score is less than 0.01. This procedure is repeated ten times, and the mode among the ten <span class="math inline">\(h\)</span>’s is picked.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">h_hat &lt;-<span class="st"> </span><span class="kw">htrain</span>(HiCR1, HiCR2, <span class="dv">1000000</span>, <span class="dt">lbr =</span> <span class="dv">0</span>, <span class="dt">ubr =</span> <span class="dv">5000000</span>, <span class="dt">range =</span> <span class="dv">0</span><span class="op">:</span><span class="dv">2</span>)</code></pre></div>
<pre><code>## smoothing:0</code></pre>
<pre><code>## smoothing:1</code></pre>
<pre><code>## smoothing:2</code></pre>
<pre><code>## Warning in htrain(HiCR1, HiCR2, 1e+06, lbr = 0, ubr = 5e+06, range = 0:2): Note: It's likely that your searching range is too 
##                         narrow. Try to expand the range and rerun it</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">h_hat</code></pre></div>
<pre><code>## [1] 1</code></pre>
<p>[Figure3. Select the optimal smoothing parameter <span class="math inline">\(h\)</span>] (<a href="https://raw.githubusercontent.com/MonkeyLB/hicrep/master/vignettes/Smoothing_parameter.JPG" class="uri">https://raw.githubusercontent.com/MonkeyLB/hicrep/master/vignettes/Smoothing_parameter.JPG</a>)</p>
<p>The above graph shows the change of SCC as the <span class="math inline">\(h\)</span> increases from 0 to 10 for a 40Kb resolution matrix. The parameter <span class="math inline">\(h = 5\)</span> is selected as the optimal smoothing neighborhood size.</p>
<p>Important note: The smoothing parameter selection could be confounded by the sequencing depth. Insufficient sequencing depth data might lead to inflated smoothing neighborhood size. To compare SCC between pairs of replicates that has the same resolution, one shall use the same smoothing parameter.</p>
<p>Train the smoothing parameter could be time-consuming. It is not suggested to train <span class="math inline">\(h\)</span> every time when calculating SCC. For a giving resolution, one could use a deeply sequenced biological replicates to train <span class="math inline">\(h\)</span> (i.e., &gt; 300 million total nubmer of reads for whole chromosome), and use the trained <span class="math inline">\(h\)</span> for other same resolution data. Here we provide some trained <span class="math inline">\(h\)</span> trained based on two replicates of hESC cells from Dixon et al 2015 (GEO accession: GSE52457):</p>
<p>Resolution (<span class="math inline">\(h\)</span>): 10K (20), 25K (10), 40k(5), 100k(3), 500k(1 or 2), 1M(0 or 1).</p>
</div>
<div id="equalize-the-total-number-of-reads" class="section level2">
<h2>Equalize the total number of reads</h2>
<p>In previous section, we mention that sequencing depth could be a confounding effect. If the total numbers of reads are very different between the two replicates, it’s suggested that one should down-sample the higher sequencing depth to make it equal to the lower one. The best way to do it is to use the bam files to do the sub-sampling randomly. In case you only have the matrix file available, we made a function <code>depth.adj()</code> to do down-sampling from matrix files.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#check total number of reads before adjustment</span>
<span class="kw">sum</span>(HiCR1)</code></pre></div>
<pre><code>## [1] 2065436</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># sub-sample 200000 total reads</span>
DS_HiCR1 &lt;-<span class="st"> </span><span class="kw">depth.adj</span>(HiCR1, <span class="dv">200000</span>) 

<span class="co">#check total number of reads after adjustment</span>
<span class="kw">sum</span>(DS_HiCR1)</code></pre></div>
<pre><code>## [1] 200035</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 3.5.0 (2018-04-23)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13.6
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] hicrep_1.5.0
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.3        compiler_3.5.0    prettyunits_1.0.2
##  [4] remotes_2.1.0     tools_3.5.0       testthat_2.3.0   
##  [7] digest_0.6.22     pkgbuild_1.0.6    pkgload_1.0.2    
## [10] memoise_1.1.0     evaluate_0.14     rhdf5_2.26.2     
## [13] rlang_0.4.1       cli_1.1.0         rstudioapi_0.10  
## [16] curl_4.2          yaml_2.2.0        xfun_0.10        
## [19] withr_2.1.2       stringr_1.4.0     strawr_0.0.1     
## [22] knitr_1.25        desc_1.2.0        fs_1.3.1         
## [25] devtools_2.2.1    rprojroot_1.3-2   glue_1.3.1       
## [28] R6_2.4.0          processx_3.4.1    rmarkdown_1.16   
## [31] sessioninfo_1.1.1 Rhdf5lib_1.4.3    callr_3.3.2      
## [34] magrittr_1.5      backports_1.1.5   ps_1.3.0         
## [37] ellipsis_0.3.0    htmltools_0.4.0   usethis_1.5.1    
## [40] assertthat_0.2.1  stringi_1.4.3     crayon_1.3.4</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#check total number of reads before adjustment</span>
<span class="kw">sum</span>(HiCR1)</code></pre></div>
<pre><code>## [1] 2065436</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># sub-sample 200000 total reads</span>
DS_HiCR1 &lt;-<span class="st"> </span><span class="kw">depth.adj</span>(HiCR1, <span class="dv">200000</span>) 

<span class="co">#check total number of reads after adjustment</span>
<span class="kw">sum</span>(DS_HiCR1)</code></pre></div>
<pre><code>## [1] 199881</code></pre>
</div>
<div id="computation-efficiency" class="section level2">
<h2>Computation efficiency</h2>
<p>Given a pair of contact maps of human chromosome 1 with bin-size equal to 40kb, it takes 27 seconds on a laptop with 2.6GHz Intel Core i7-6600U and 16Gb of RAM.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
